File: Tree/Traverals/C++/inOrder.cpp
#include <iostream>
#include <queue>
using namespace std;

struct Node{
    //This is the structure of the binary tree.
    int data;
    Node *left;
    Node *right;
};

void inOrder(Node *root)
{
	if(root!=NULL)
	{
		inOrder(root->left);
		cout << root->data << " ";
		inOrder(root->right);
	}
}


int main(int argc, char const *argv[])
{
    // Initializing a binary tree.
    Node *root = new Node();
    root->data = 10;
    root->right = new Node();
    root->right->data = 15;
    root->left = new Node();
    root->left->data = 5;
    root->left->left = new Node();
    root->left->left->data = 30;
    root->left->right = new Node();
    root->left->right->data = 50;

    inOrder(root);
    return 0;
}    

File: Tree/Traverals/JAVA/inorder.java
/*when you are compiling your program change the .java class file to inorder because the class declared as public*/

import java.io.*;
import java.util.*;
 public class inorder{
 
   private Node root;
   private class Node{
      
     private Node left;
     private Node right;
     int data;
      public Node(int data){
        this.data=data;
        System.out.print(data+"\t");
     }
  }

 public void createBinaryTree(){
       
       Node first = new Node(1);
       Node second = new Node(2);
       Node third = new Node(3);
       Node fourth = new Node(4);
       Node fifth = new Node(5);

        root = first;
        first.left = second;
        first.right = third;
        second.left = fourth;
        second.right = fifth;
     }

 /* public void inorders(Node root){

         if(root == null){
        return ; 
        }
       inorders(root.left);
       System.out.print(root.data+"\t");
       inorders(root.right);
   } */

/* this is using the stack concept*/

  public void inorders(){

  if(root == null){
    return; 
  }
 Stack<Node> stack = new Stack<>();
  Node temp = root;
  while(!stack.isEmpty() || (temp!=null)){
     if(temp!=null){
      stack.push(temp);
      temp = temp.left;
      }else{
        temp = stack.pop();
        System.out.print(temp.data+"\t");
        temp = temp.right;
      }
   }  
}

 public static void main(String args[]){

   inorder obj = new inorder();
   System.out.println("Before inorder");
   obj.createBinaryTree();
   System.out.println(); 
   System.out.println("After inorder");
   obj.inorders();
  }
}